### PR TITLE
Server lease

### PR DESCRIPTION
**Pull request by ffilippopoulos**

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	Name  string             `json:"name"`
+	Peers *[]AgentPeerConfig `json:"peers"`
+}
+
+type AgentConfig struct {
+	Oidc *AgentOidcConfig  `json:"oidc"`
+	Devs *[]AgentDevConfig `json:"devs"`
+}
+
+func UnmarshallAgentConfig(data []byte, v interface{}) error {
```
Comment (alkar) [inline config.go:30]: nitpick: do we really need this alias?
Not really opposed to it, it just feels more natural to use `json.Unmarshal` directly.
Also Unmarshal is with a single l )

File: config.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	TokenUrl string `json:"tokenUrl"`
+}
+
+type AgentPeerConfig struct {
+	Url string `json:"url"`
+}
+
+type AgentDevConfig struct {
+	Name  string             `json:"name"`
+	Peers *[]AgentPeerConfig `json:"peers"`
```
Comment (alkar) [inline config.go:22]: This is a bit strange, a pointer to a slice. Why not just `[]AgentPeerConfig` ?

File: config.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+type AgentConfig struct {
+	Oidc *AgentOidcConfig  `json:"oidc"`
+	Devs *[]AgentDevConfig `json:"devs"`
```
Comment (alkar) [inline config.go:27]: Same as above. As far as I can tell, you always dereference these to use them.

File: config.go (hunk 1 above)
Comment (ffilippopoulos) [reply config.go:30]: tbh, that alias was handy for me to test reading the config while writing this file. Fixing the typo

**Pull request merged**